### PR TITLE
Artifactory Tab: Add build-scan report icon and link

### PR DESCRIPTION
**Pull request by barbelity**

### CODE REVIEW

File: scripts/build-info.js [hunk 1]
```diff
@@ -58,4 +79,30 @@ define(["TFS/DistributedTask/TaskRestClient"], (taskRestClient) => {
...
+     */
+    function createXrayScanDiv(scanUrl) {
+        let xrayScanDiv = document.createElement('div');
+        let xrayScanIcon = document.createElement('img');
+        let xrayScanUrlDiv = document.createElement('a');
+
+        xrayScanIcon.src = "images/artifactory-xray-scan.png";
+        xrayScanUrlDiv.classList.add("xray-scan-url");
+        xrayScanUrlDiv.href = scanUrl;
+        xrayScanUrlDiv.text = "Build Scan Report";
```
Comment (eyalbe4) [inline scripts/build-info.js]: Build Scan Report --> Xray Build Scan Report

File: tasks/ArtifactoryXrayScan/xrayScan.js [hunk 2]
```diff
@@ -18,9 +19,21 @@ function RunTaskCbk(cliPath) {
...
+
+        // Save build-scan link in file.
+        let workDir = tl.getVariable('System.DefaultWorkingDirectory');
+        if (!workDir) {
+            tl.setResult(tl.TaskResult.Failed, "Failed getting default working directory.");
+            return;
+        }
+        let scanUrlFile = path.join(workDir, "scanUrlFile");
+        tl.writeFile(scanUrlFile, JSON.stringify(scanUrl));
+        console.log("##vso[task.addattachment type=xrayType;name=scanDetails;]" + scanUrlFile);
```
Comment (eyalbe4) [inline tasks/ArtifactoryXrayScan/xrayScan.js]: Question - do we need the file path to be added as a log message?

File: tasks/ArtifactoryXrayScan/xrayScan.js [hunk 3]
```diff
@@ -18,9 +19,21 @@ function RunTaskCbk(cliPath) {
 
     try {
         let taskRes = utils.executeCliCommand(cliCommand, process.cwd(), 'pipe');
-        console.log(taskRes);
-        // Check if should fail build.
+        console.log("Scan result:\n" + taskRes);
         let resJson = JSON.parse(taskRes);
+        let scanUrl = resJson['summary']['more_details_url'];
+
+        // Save build-scan link in file.
```
Comment (eyalbe4) [inline tasks/ArtifactoryXrayScan/xrayScan.js]: in file. --> to a file.

File: tasks/ArtifactoryXrayScan/xrayScan.js (hunk 2 above)
Comment (barbelity) [reply tasks/ArtifactoryXrayScan/xrayScan.js]: This line doesn't actually produces a log message, it is Azure DevOps' way to store the information we want to present in the 'Artifactory' tab.

**Pull request merged**